server/rest/postgres_geojson.py

```diff
@@ -0,0 +1,368 @@
+        return resp
+
+    @access.user
+    @loadmodel(model='assetstore', map={'assetstoreId': 'assetstore'})
+    @describeRoute(
+        Description('Create json dataset for the given view/table filtering values')
+        .param('assetstoreId', 'assetstore ID of the target database')
+        .param('table', 'Table name from the database')
+        .param('field', 'Field to which the aggregate function will be applied')
+        .param('aggregationFunction', 'aggregate function used on the field')
+        .param('filter', 'Filter condition object for filtering table data')
+        .param('geometryField', 'Geometry data definition object')
+        .param('datasetName', 'A custom name for the dataset', required=False)
+    )
+    def createPostgresGeojsonData(self, assetstore, params):
+        filter = params['filter']
+        table = params['table']
+        field = params['field']
+        aggregateFunction = params['aggregateFunction']
+        geometryField = json.loads(params['geometryField'])
+
+        if geometryField['type'] == 'built-in':
+            buildInGeomField = geometryField['field']
+            properties = [field, {
+                'func': aggregateFunction,
+                'param': {'field': field}
+            }]
+            # add string fields with concat aggregate function and in the format for
+            # json_build_object
+            for property in [(i['name'], {
+                'func': 'string_agg',
+                'param': [{
+                    'func': 'distinct',
+                    'param': {'field': i['name']}
+                }, '|'],
+                'reference': i['name']
+            })
+                    for i in self._getColumns(assetstore, {'table': params['table']})
+                    if i['datatype'] == 'string'
+                    if i['name'] != field]:
+                properties.extend(property)
+            fields = [{
+                'func': 'json_build_object', 'param': [
+                    'type', 'Feature',
+                    'geometry', {
+                        'func': 'cast', 'param': [{
+                            'func': 'st_asgeojson', 'param': [{
+                                'func': 'st_transform', 'param': [{'field': buildInGeomField}, 4326]
+                            }]}, 'JSON']
+                    },
+                    'properties', {
+                        'func': 'json_build_object', 'param': properties
+                    }
+                ]
+            }]
+            group = [buildInGeomField]
+        elif geometryField['type'] == 'link':
+            fields = [{
+                'func': aggregateFunction,
+                'param': {'field': field},
+                'reference': field
+            }]
+            group = [x['value'] for x in geometryField['links']]
+            # add string fields with concat aggregate function
+            for i in [i for i in self._getColumns(assetstore, {'table': params['table']})
```

Comment:
Here you are looping through a list that is created by a list comprehension. I think you don't need the list comprehension here. IMHO you should either do a for loop or list comprehension but not both. I believe this is too complicated to be a list comprehension and will be easier to read if it is just a for loop.